public/components/trace_analytics/requests/traces_request_handler.ts

```diff
@@ -157,8 +158,37 @@ export const handleTracesRequest = async (
         })
       : Promise.resolve({});
 
-  return Promise.allSettled([responsePromise, percentileRangesPromise])
-    .then(([responseResult, percentileRangesResult]) => {
+  let rootSpansTotalHitsPromise: Promise<number> | undefined;
+
+  if (setTotalHits && (mode === 'data_prepper' || mode === 'custom_data_prepper')) {
+    const rootSpansDSL = cloneDeep(timeFilterDSL);
+    rootSpansDSL.query.bool.filter.push({
```

Comment:
Looks like we are adding a `_search` request, this should be just `_count` request as we need just the hits here.

Reply:
The size 0 part prevents the more intensive aggregation part and acts similar to count.
Keeping current implementation for the time being.